src/main/java/mil/dds/anet/resources/NoteResource.java

```diff
@@ -91,9 +89,41 @@ public Integer deleteNote(@GraphQLRootContext Map<String, Object> context,
   }
 
   private void checkPermission(Note n, final Person user) {
-    if (!n.getAuthorUuid().equals(DaoUtils.getUuid(user)) && !AuthUtils.isAdmin(user)) {
-      throw new WebApplicationException("Only the author or an admin can do this",
-          Status.FORBIDDEN);
+    // Node type is ASSESSMENT
+    if (n.getType() == NoteType.ASSESSMENT && !AuthUtils.isAdmin(user)) {
+      for (NoteRelatedObject note : n.getNoteRelatedObjects()) {
+        // Task for
+        if (TaskDao.TABLE_NAME.equals(note.getRelatedObjectType())) {
+          final Task userTask = taskDao.getByUuid(note.getRelatedObjectUuid());
+          final List<Position> responsiblePositions =
+              userTask.loadResponsiblePositions(engine.getContext()).join();
+          if (Utils.isEmptyOrNull(responsiblePositions)) {
+            throw new WebApplicationException("You do not have permissions to edit this assessment",
+                Status.FORBIDDEN);
+          }
+          // checked whether the user's current position is in the list of responsiblePositions.
+          final Position userPosition = user.loadPosition();
+          if (!responsiblePositions.contains(userPosition)) {
+            throw new WebApplicationException("You do not have permissions to edit this assessment",
+                Status.FORBIDDEN);
+          }
+        }
+        // Person for
+        else if (PersonDao.TABLE_NAME.equals(note.getRelatedObjectType())) {
+          final Position userPosition = user.loadPosition();
+          final List<Position> associatedPositions =
+              userPosition.loadAssociatedPositions(engine.getContext()).join();
+          if (Utils.isEmptyOrNull(associatedPositions)) {
+            throw new WebApplicationException("You do not have permissions to edit this assessment",
+                Status.FORBIDDEN);
+          }
+          // checked whether the user's current position is in the list of associatedPositions
+          if (!associatedPositions.contains(userPosition)) {
```

Comment:
We want to check if person who is being assessed has position in associated positions of current user or vice versa. something like:
```
final Person relatedPerson = personDao.getByUuid(note.getRelatedObjectUuid());
final Position relatedPosition = relatedPerson.loadPosition();
if (userPosition.associatedPositions.contains(relatedPosition))...
```